### PR TITLE
Added implement the GetRightDiagonal method. Closed #29

### LINKED ISSUE #29: Implement the GetRightDiagonal method

### PR DESCRIPTION
**Pull request by Kasyanov-Igor**

### CODE REVIEW

**Review by artemqdh — COMMENTED** [review]
documentation?

**Review by TopAcademy2022 — CHANGES_REQUESTED** [review]
Rename commit.

File: Checkers/Logic/GameField.cs [hunk 1]
```diff
@@ -73,11 +73,51 @@ private List<KeyValuePair<char, int>> GetLeftDiagonal()
 			return result;
 		}
 
-		private List<KeyValuePair<char, int>> GetRightDiagonal()
+		private List<KeyValuePair<char, int>> GetRightDiagonal(GameFigure DirectionMovement)
```
Comment (TopAcademy2022) [inline Checkers/Logic/GameField.cs:76]: Add documentation.

File: Checkers/Logic/GameField.cs [hunk 2]
```diff
@@ -73,11 +73,51 @@ private List<KeyValuePair<char, int>> GetLeftDiagonal()
...
 		{
 			List<KeyValuePair<char, int>> result = new List<KeyValuePair<char, int>>();
 
+			KeyValuePair<char, int> presentPosition = this._checkersField[DirectionMovement];
 
+			if (!DirectionMovement.IsQueen())
```
Comment (TopAcademy2022) [inline Checkers/Logic/GameField.cs:82]: Nope.

Comment (TopAcademy2022) [reply Checkers/Logic/GameField.cs:82]: Rename parameter.

File: Checkers/Logic/GameField.cs [hunk 3]
```diff
@@ -73,11 +73,51 @@ private List<KeyValuePair<char, int>> GetLeftDiagonal()
...
+			{
+				if (DirectionMovement.GetDirectionMovement() == GameFigure.DirectionMovement.Up)
+				{
+					for (int i = 1; i < COUNT_ROWS; ++i)
+					{
+						if (presentPosition.Key + i <= 104 && presentPosition.Key + i >= 97 && (presentPosition.Value + i) <= COUNT_ROWS)
+						{
+							result.Add(new KeyValuePair<char, int>((char)(presentPosition.Key + i), presentPosition.Value + i));
+
```
Comment (TopAcademy2022) [inline Checkers/Logic/GameField.cs:91]: Remove empty line.

File: Checkers/Logic/GameField.cs [hunk 4]
```diff
@@ -73,11 +73,51 @@ private List<KeyValuePair<char, int>> GetLeftDiagonal()
...
 
+			KeyValuePair<char, int> presentPosition = this._checkersField[DirectionMovement];
 
+			if (!DirectionMovement.IsQueen())
+			{
+				if (DirectionMovement.GetDirectionMovement() == GameFigure.DirectionMovement.Up)
+				{
+					for (int i = 1; i < COUNT_ROWS; ++i)
+					{
+						if (presentPosition.Key + i <= 104 && presentPosition.Key + i >= 97 && (presentPosition.Value + i) <= COUNT_ROWS)
```
Comment (TopAcademy2022) [inline Checkers/Logic/GameField.cs:88]: Remove magic numbers.

File: Checkers/Logic/GameField.cs [hunk 5]
```diff
@@ -73,11 +73,51 @@ private List<KeyValuePair<char, int>> GetLeftDiagonal()
...
+							result.Add(new KeyValuePair<char, int>((char)(presentPosition.Key + i), presentPosition.Value + i));
+
+						}
+					}
+				}
+				else if (DirectionMovement.GetDirectionMovement() == GameFigure.DirectionMovement.Down)
+				{
+					for (int i = 1; i < COUNT_ROWS; ++i)
+					{
+						if (presentPosition.Key - i <= 104 && presentPosition.Key - i >= 97 && presentPosition.Value - i >= 1)
```
Comment (TopAcademy2022) [inline Checkers/Logic/GameField.cs:99]: See 88 line.

File: Checkers/Logic/GameField.cs [hunk 6]
```diff
@@ -73,11 +73,51 @@ private List<KeyValuePair<char, int>> GetLeftDiagonal()
...
+							result.Add(new KeyValuePair<char, int>((char)(presentPosition.Key - i), presentPosition.Value - i));
+						}
+					}
+				}
+			}
+			else
+			{
+				for (int i = 1; i < COUNT_ROWS; ++i)
+				{
+					if (presentPosition.Key + i <= 104 && presentPosition.Key + i >= 97 && (presentPosition.Value + i) <= COUNT_ROWS)
```
Comment (TopAcademy2022) [inline Checkers/Logic/GameField.cs:110]: See again 88 line.

File: Checkers/Logic/GameField.cs [hunk 7]
```diff
@@ -73,11 +73,51 @@ private List<KeyValuePair<char, int>> GetLeftDiagonal()
...
+					{
+						result.Add(new KeyValuePair<char, int>((char)(presentPosition.Key + i), presentPosition.Value + i));
+
+					}
+					if (presentPosition.Key - i <= 104 && presentPosition.Key - i >= 97 && presentPosition.Value - i >= 1)
```
Comment (TopAcademy2022) [inline Checkers/Logic/GameField.cs:115]: See 88 line (magic numbers).

File: Checkers/Logic/GameField.cs [hunk 8]
```diff
@@ -73,11 +73,51 @@ private List<KeyValuePair<char, int>> GetLeftDiagonal()
...
+					for (int i = 1; i < COUNT_ROWS; ++i)
+					{
+						if (presentPosition.Key - i <= 104 && presentPosition.Key - i >= 97 && presentPosition.Value - i >= 1)
+						{
+							result.Add(new KeyValuePair<char, int>((char)(presentPosition.Key - i), presentPosition.Value - i));
+						}
+					}
+				}
+			}
+			else
```
Comment (TopAcademy2022) [inline Checkers/Logic/GameField.cs:106]: Set as if(IsQueen()).

File: Checkers/Logic/GameField.cs [hunk 9]
```diff
@@ -73,11 +73,51 @@ private List<KeyValuePair<char, int>> GetLeftDiagonal()
...
-		private List<KeyValuePair<char, int>> GetRightDiagonal()
+		private List<KeyValuePair<char, int>> GetRightDiagonal(GameFigure DirectionMovement)
 		{
 			List<KeyValuePair<char, int>> result = new List<KeyValuePair<char, int>>();
 
+			KeyValuePair<char, int> presentPosition = this._checkersField[DirectionMovement];
 
+			if (!DirectionMovement.IsQueen())
+			{
+				if (DirectionMovement.GetDirectionMovement() == GameFigure.DirectionMovement.Up)
```
Comment (TopAcademy2022) [inline Checkers/Logic/GameField.cs:84]: Replace, see 106 line.

File: Checkers/Logic/GameField.cs [hunk 10]
```diff
@@ -73,11 +73,51 @@ private List<KeyValuePair<char, int>> GetLeftDiagonal()
...
+				}
+			}
+			else
+			{
+				for (int i = 1; i < COUNT_ROWS; ++i)
+				{
+					if (presentPosition.Key + i <= 104 && presentPosition.Key + i >= 97 && (presentPosition.Value + i) <= COUNT_ROWS)
+					{
+						result.Add(new KeyValuePair<char, int>((char)(presentPosition.Key + i), presentPosition.Value + i));
+
```
Comment (TopAcademy2022) [inline Checkers/Logic/GameField.cs:113]: Remove empty line.

File: Checkers/Logic/GameField.cs (hunk 9 above)
Comment (TopAcademy2022) [inline Checkers/Logic/GameField.cs:84]: Move DirectionMovement.GetDirectionMovement in variable.

**Review by TopAcademy2022 — CHANGES_REQUESTED** [review]

**Review by Viktor080181 — APPROVED** [review]

**Pull request closed without merging**